### PR TITLE
ROX-14061, ROX-14277: collection deletion error message, default sorting options

### PR DESCRIPTION
**Pull request by keyallis**
## Description

Made the deletion error more human readable on a failed deletion.
Added default sort option of Collection Name for the collections List endpoint.
Added default sort option of Deployment Name for the deployment matching feature of various endpoints.

## Checklist
- [ ] Investigated and inspected CI test results

## Testing Performed

Manually verified the error message via FE

### CODE REVIEW

File: central/resourcecollection/datastore/datastore_impl.go [hunk 1]
```diff
@@ -409,7 +409,10 @@ func (ds *datastoreImpl) DeleteCollection(ctx context.Context, id string) error
 	// delete from storage first so postgres can tell if the collection is referenced by another
 	err = ds.storage.Delete(ctx, id)
 	if err != nil {
-		return err
+		if strings.Contains(err.Error(), "SQLSTATE 23503") {
+			return errors.Wrap(errox.ReferencedByAnotherObject, "failed to delete collection")
+		}
+		return errors.Wrap(err, "failed to delete collection")
```
Comment (md2119) [inline central/resourcecollection/datastore/datastore_impl.go:415]:
```suggestion
		if strings.Contains(err.Error(), "SQLSTATE 23503") {
			err = errox.ReferencedByAnotherObject
		}
		return errors.Wrap(err, "failed to delete collection")
```

**Review by md2119 — APPROVED** [review]

**Review by dvail — APPROVED** [review]
👍  Here's how it looks in the UI:
image

**Pull request merged**